python/cudf/bindings/dlpack.pyx

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+from cudf.bindings.cudf_cpp cimport *
+from cudf.bindings.cudf_cpp import *
+from librmm_cffi import librmm as rmm
+
+from libc.stdint cimport uintptr_t, int8_t
+from libc.stdlib cimport calloc, malloc, free
+from cpython cimport pycapsule
+
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+
+
+cpdef from_dlpack(dlpack_capsule):
+    """
+    Converts a DLPack Tensor PyCapsule into a list of cudf Column objects.
+    """
+    cdef DLManagedTensor_* dlpack_tensor = pycapsule.PyCapsule_GetPointer(dlpack_capsule, 'dltensor')
+    cdef gdf_size_type c_result_num_cols
+    cdef gdf_column* result_cols
+
+    with nogil:
+        result = gdf_from_dlpack(
+            &result_cols,
+            &c_result_num_cols,
+            dlpack_tensor
+        )
+
+    check_gdf_error(result)
+    
+    # TODO: Replace this with a generic function for cudf_cpp.pyx since this is
+    # copied from join.pyx
+    res = []
+    valids = []
+    for idx in range(c_result_num_cols):
+        data_ptr = <uintptr_t>result_cols[idx].data
+        res.append(rmm.device_array_from_ptr(ptr=data_ptr,
+                                             nelem= result_cols[idx].size,
+                                             dtype=gdf_to_np_dtype( result_cols[idx].dtype),
+                                             finalizer=rmm._make_finalizer(
+                                                 data_ptr, 0)))
+        valid_ptr = <uintptr_t>result_cols[idx].valid
+        valids.append(rmm.device_array_from_ptr(ptr=valid_ptr,
+                                                nelem=calc_chunk_size(
+                                                    result_cols[idx].size, mask_bitsize),
+                                                dtype=mask_dtype,
+                                                finalizer=rmm._make_finalizer(
+                                                    valid_ptr, 0)))
+    return res, valids
+
+
+cpdef to_dlpack(in_cols):
+
```

Comment:
Missing docstring. Suggest documenting the input and output, especially the data type and name that will be packaged in the PyCapsule.